syft/frameworks/torch/tensors/interpreters/large_precision.py

```diff
@@ -0,0 +1,175 @@
+import numpy as np
+import torch
+
+from syft.frameworks.torch.overload_torch import overloaded
+from syft.frameworks.torch.tensors.interpreters import AbstractTensor
+
+
+class LargePrecisionTensor(AbstractTensor):
+    """LargePrecisionTensor allows handling of numbers bigger than LongTensor
+
+    Some systems using Syft require larger types than those supported natively. This tensor type supports arbitrarily
+    large values by packing them in smaller ones.
+    """
+
+    def __init__(
+        self, owner=None, id=None, tags=None, description=None, precision=16, virtual_prec=0
+    ):
+        """Initializes a LargePrecisionTensor.
+
+        Args:
+            owner (BaseWorker): An optional BaseWorker object to specify the worker on which
+                the tensor is located.
+            id (str or int): An optional string or integer id of the LargePrecisionTensor.
+            tags (list): list of tags for searching.
+            description (str): a description of this tensor.
+            precision (int): The precision this tensor will be transformed to internally. In bits.
+            virtual_prec (int): The virtual precision required by the caller. In bits.
+        """
+        super().__init__(id=id, owner=owner, tags=tags, description=description)
+        self.precision = precision
+        self.virtual_prec = virtual_prec
+
+    @staticmethod
+    def _create_internal_tensor(tensor, precision, virtual_prec):
+        # TODO refs_ok might not be needed as we are passing now torch.tensors here
+        # refs_ok is necessary to enable iterations of reference types.
+        # These big numbers are stored as objects in np
+        result = []
+        for x in np.nditer(tensor, flags=["refs_ok"]):
+            n = int(x.item() * 2 ** virtual_prec)
+            print(f"\nAdding number {n} for item {x.item()}\n")
+            result.append(LargePrecisionTensor._split_number(n, precision))
+        new_shape = tensor.shape + (len(max(result, key=len)),)
+        result = np.array(result).reshape(new_shape)
+        return torch.IntTensor(result)
+
+    def get_class_attributes(self):
+        """
+        Specify all the attributes need to build a wrapper correctly when returning a response.
+        """
+        return {"precision": self.precision, "virtual_prec": self.virtual_prec}
+
+    def __eq__(self, other):
+        return self.child == other.child
+
+    @overloaded.method
+    def add(self, self_, *args, **kwargs):
+        other = args[0]
+        if self_.shape[-1] != other.shape[-1]:
+            result = self._add_different_dims(other, self_)
+        else:
+            result = self_ + other
+        return LargePrecisionTensor(precision=self.precision, virtual_prec=self.virtual_prec).on(
+            result
+        )
```

Comment:
The @overloaded.method decorator will automatically put LargePrecisionTensor on top of your result, so you can remove it, otherwise you have it twice (as cell 17 of the NB you shared shows)
Also, you can simplify the function signature `def add(self, self_, *args, **kwargs)` -> `def add(self, self_, other)` directly